backend/app/services/database.py

```
@@ -199,385 +199,357 @@ def _initialize_database(self):
-                    conn.execute('COMMIT')
-                    
-                    # Remove from cache if exists
-                    self.script_cache.delete(script_id)
-                    
-                    return deleted
-                except Exception as e:
-                    conn.execute('ROLLBACK')
-                    raise
         except Exception as e:
-            logger.error(f"Failed to delete script: {str(e)}")
+            logger.error(f"Failed to update script: {str(e)}")
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)